roscala/src/main/scala/coop/rchain/rosette/Queue.scala

```diff
@@ -0,0 +1,101 @@
+package coop.rchain.rosette
+
+case class Queue(elems: Tuple) extends Ob {
+
+  def head: Ob = elems.elem.head
+
+  def tail: Ob = elems.elem.last
+
+  def nElems: Ob = Fixnum(elems.elem.length)
+
+  def depth(): Int =
+    elems.elem.length
+
+  def isEmpty(): Boolean =
+    (nElems.asInstanceOf[Fixnum] == Fixnum(0)).value
+
+  def enqueue(ele: Ob): Queue =
+    Queue(Tuple(elems.elem :+ ele))
```

Comment:
Are you sure you want prepend here? The C++ enqueue method seems to be doing an append.

Reply:
The ':+' operator in scala means doing an append, i think it's right to use here.

Reply:
You're right.  I read it wrong.